pyo3-ffi/src/cpython/listobject.rs

```diff
@@ -28,9 +28,18 @@ pub unsafe fn PyList_GET_ITEM(op: *mut PyObject, i: Py_ssize_t) -> *mut PyObject
 
 /// Macro, *only* to be used to fill in brand new lists
 #[inline]
-#[cfg(not(any(PyPy, GraalPy)))]
+#[cfg(not(PyPy))]
 pub unsafe fn PyList_SET_ITEM(op: *mut PyObject, i: Py_ssize_t, v: *mut PyObject) {
-    *(*(op as *mut PyListObject)).ob_item.offset(i) = v;
+    #[cfg(not(GraalPy))]
+    {
+        *(*(op as *mut PyListObject)).ob_item.offset(i) = v;
+    }
+    #[cfg(GraalPy)]
+    {
+        let previous = crate::PyList_GetItem(op, i);
+        Py_XDECREF(previous);
```

Comment:
This seems wrong. The whole point of `PyList_SET_ITEM` is that it doesn't decref the existing items because they are not initialized. Here you even decref it twice, once manually, second time in `PyList_SetItem`. What was wrong with using `_PyList_SET_ITEM`?

Reply:
Right, I confused myself